src/App.js

```diff
@@ -40,31 +41,47 @@ const people = [
 const headers = ["First Name", "Last Name"];
 
 const App = () => {
-  const { selected, handleSelectOne, handleSelectAll } = useRowSelection(
-    people.length
-  );
+  const selectedValues = useRef({});;
+  const { 
+    isToggled: allValuesSelected, 
+    toggle 
+  } = useToggle();
 
   return (
     <>
       <Table>
         <Header>
-          <SelectableRow onChange={handleSelectAll}>
-            {headers.map((header, index) => {
-              return <HeaderCell key={index}>{header}</HeaderCell>;
-            })}
-          </SelectableRow>
+          <BaseSelectableRow onChange={toggle} checked={allValuesSelected} >
```

Comment:
Changed the hooks so we can borrow functionality from SelectableRow. If all of the values are selected we don't need to add them anywhere, instead, we can refer back to the initial data set for usage when needed